src/sentry/tasks/post_process.py

```diff
@@ -1189,7 +1190,21 @@ def process_service_hooks(job: PostProcessJob) -> None:
         if has_alert:
             allowed_events.add("event.alert")
 
-        if allowed_events:
+        if in_random_rollout("process_service_hook.payload.rollout"):
+            payload = None
+            for servicehook_id, events in _get_service_hooks(project_id=event.project_id):
+                if any(e in allowed_events for e in events):
+                    # only generate the payload once to reduce snuba queries
+                    if payload is None:
+                        # TODO: when we have multiple service hook versions, this will need to change.
+                        payload = json.dumps(get_payload_v0(event))
```

Comment:
Is there a way to serialize the event without doing this call up front? Then we could move all this to the async task function?

Reply:
There should be a way to get the event/project/group id, and then use that to rebuild the event in the webhook task. For projects with multiple hooks, we'll read the event multiple times though.

Reply:
Updated to pass project_id, group_id, and event_id. This approach moves serialization of the event until the hook is delivered. We have to re-read the event from nodestore, but that read *should* be cached most of the time as the event was recently in post_process.

I've also checked that all the events that go to service hooks are error events, so we don't have to worry about issue occurrences. 🤞